### PR TITLE
[prometheus-redis-exporter] Allow `REDIS_ADDR` environment variable to be configured through a secret

### PR DESCRIPTION
**Pull request by daconstenla**
#### What this PR does / why we need it

This pr adds support for (prometheus-redis-exporter)[./charts/prometheus-redis-exporter/] redis address environment variable to be configured from a secret.
It would unlock for redis address to be fetched though eg: [external-secrets](https://external-secrets.io/).

#### Which issue this PR fixes

There is no issue linked to this change (that I know of).

#### Special notes for your reviewer

I've performed some sanity checks locally to verify the chart configuration stays as is and the new configuration also works.
Since changes are all around deployment's configuration I'm using `helm` + `yq` to focus on that part of the rendered template.

This is the specific used command:
```bash
helm template . -f values.yaml |yq 'select(.kind == "Deployment").spec.template.spec.containers[0].env'
```

##### using redisAddressConfig.enabled = false

No diff, it's literally `values.yaml` by default, renders as:

```yaml
- name: REDIS_ADDR
  value: redis://myredis:6379
```

##### using redisAddressConfig.enabled = true + redisAddressConfig.isASecret = false

```diff
 redisAddressConfig:
   # configure `REDIS_ADDR` from a configmap
-  enabled: false
+  enabled: true
   # if `true` the `REDIS_ADDR` is sourced on a secret instead of a configmap
   isASecret: false
   # Use existing configmap (ignores redisAddress)
   source:
-    name: ""
-    key: ""
+    name: "testName"
+    key: "testKey"
```
Renders as:
```yaml
- name: REDIS_ADDR
  valueFrom:
    configMapKeyRef:
      name: testName
      key: testKey
```
##### using redisAddressConfig.enabled = true + redisAddressConfig.isASecret = true

```diff
 redisAddressConfig:
   # configure `REDIS_ADDR` from a configmap
-  enabled: false
+  enabled: true
   # if `true` the `REDIS_ADDR` is sourced on a secret instead of a configmap
-  isASecret: false
+  isASecret: true
   # Use existing configmap (ignores redisAddress)
   source:
-    name: ""
-    key: ""
+    name: "testName"
+    key: "testKey"
```
renders as:
```yaml
- name: REDIS_ADDR
  valueFrom:
    secretKeyRef:
      name: testName
      key: testKey
```

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zanhsieh — APPROVED** [review]

**Comment by daconstenla** [comment]
@zanhsieh should I do anything else to be able to merge this change? 🙏

**Comment by zanhsieh** [comment]
> @zanhsieh should I do anything else to be able to merge this change? 🙏

Nothing. Just give another maintainer @acondrat some time before I merging this.

File: charts/prometheus-redis-exporter/Chart.yaml
Comment (zeritti) [inline charts/prometheus-redis-exporter/Chart.yaml:5]: Removing field `redisAddressConfig.configmap` makes the chart incompatible for the users making use of the field - an upgrade will fail without a prior adjustment. In this respect, the change is major deserving an upgrade note in README.

```suggestion
version: 6.0.0
```

File: charts/prometheus-redis-exporter/templates/deployment.yaml
Comment (zeritti) [inline charts/prometheus-redis-exporter/templates/deployment.yaml:58]: I'd suggest to remove the indefinite article.

```suggestion
              {{- if .Values.redisAddressConfig.isSecret }}
```

File: charts/prometheus-redis-exporter/values.yaml
Comment (zeritti) [inline charts/prometheus-redis-exporter/values.yaml:68]: This comment may need an update.

File: charts/prometheus-redis-exporter/values.yaml
Comment (zeritti) [inline charts/prometheus-redis-exporter/values.yaml:67]:
```suggestion
  isSecret: false
```

**Comment by daconstenla** [comment]
Thank you so much for the comments @zanhsieh, good point.

Accepted the suggestions and changed the outdated comments, I hope they are good 👌.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**
